pkg/utils/logger/logger.go

```diff
@@ -13,69 +13,115 @@
 
 package logger
 
-import (
-	"fmt"
-	"os"
-	"strings"
-
-	log "github.com/cihub/seelog"
-)
-
 const (
-	envLogLevel    = "AWS_VPC_K8S_CNI_LOGLEVEL"
-	envLogFilePath = "AWS_VPC_K8S_CNI_LOG_FILE"
-	// logConfigFormat defines the seelog format, with a rolling file
-	// writer. We cannot do this in code and have to resort to using
-	// LoggerFromConfigAsString as seelog doesn't have a usable public
-	// implementation of NewRollingFileWriterTime
-	logConfigFormat = `
-<seelog type="asyncloop" minlevel="%s">
- <outputs formatid="main">
-  %s
- </outputs>
- <formats>
-  <format id="main" format="%%UTCDate(2006-01-02T15:04:05.000Z07:00) [%%LEVEL] %%t%%Msg%%n" />
- </formats>
-</seelog>
-`
+	envLogLevel              = "AWS_VPC_K8S_CNI_LOGLEVEL"
+	defaultIPamDLogFilePath  = "/host/var/log/aws-routed-eni/ipamd.log"
+	envIPamDLogFilePath      = "AWS_VPC_K8S_CNI_LOG_FILE"
+	defaultPluginLogFilePath = "/var/log/aws-routed-eni/plugin.log"
+	envPluginFilePath        = "AWS_VPC_K8S_PLUGIN_LOGL_FILE"
```

Comment:
I would expect `AWS_VPC_K8S_PLUGIN_LOGL_FILE` to be passed to the plugin, just [like](https://github.com/aws/amazon-vpc-cni-k8s/blob/8340e95126da40e8489074c68052869c4cd098d2/scripts/entrypoint.sh#L76) the [`MTU`](https://github.com/aws/amazon-vpc-cni-k8s/blob/8340e95126da40e8489074c68052869c4cd098d2/cmd/routed-eni-cni-plugin/cni.go#L71). The ENV variables in the daemonset config is only available to ipamd.

Also, what does logging to `stdout` mean here? It will end up in the kubelet logs? Have we tested this config?

Reply:
> I would expect `AWS_VPC_K8S_PLUGIN_LOGL_FILE` to be passed to the plugin, just [like](https://github.com/aws/amazon-vpc-cni-k8s/blob/8340e95126da40e8489074c68052869c4cd098d2/scripts/entrypoint.sh#L76) the [`MTU`](https://github.com/aws/amazon-vpc-cni-k8s/blob/8340e95126da40e8489074c68052869c4cd098d2/cmd/routed-eni-cni-plugin/cni.go#L71). The ENV variables in the daemonset config is only available to ipamd.

I did not understand this point. Are you saying that the environment variable configured with `config.yaml` are only available for `L-IPamD` server ? I believe, to override the current `MTU` value, we still pass `AWS_VPC_ENI_MTU=9001`, right ? Please correct me if I am missing something here. 

> Also, what does logging to `stdout` mean here? It will end up in the kubelet logs? Have we tested this config?

`stdout` here write to default container logging i.e., `stdout/stderr`. Yes, I have tested it, but found another issue with the `entrypoint.sh` [script](https://github.com/aws/amazon-vpc-cni-k8s/blob/8340e95126da40e8489074c68052869c4cd098d2/scripts/entrypoint.sh#L87) where the logs which are written to stdout/stderr are piped to a file which is `cat`'ed out only upon `L-IPamD` initiation failure. (I will put in a new PR later for the `entrypoint.sh` script)

Reply:
Yes, ENV variables set in `aws-k8s-cni.yaml` are only available to ipamd. The MTU values get set in the plugin config file so that the plugin can get it. See the linked examples in the previous comment.